futures-executor/src/local_pool.rs

```diff
@@ -132,35 +132,31 @@ impl LocalPool {
     /// be inert after the call completes, but can continue with further use of
     /// `run` or `run_until`. While the function is running, however, all tasks
     /// in the pool will try to make progress.
-    pub fn run_until<F>(&mut self, mut f: F, exec: &mut Executor) -> Result<F::Item, F::Error>
-        where F: Future
+    pub fn run_until<F, Exec>(&mut self, mut f: F, exec: &mut Exec) -> F::Output
+        where F: Future + Unpin, Exec: Executor + Sized
```

Comment:
This `Unpin` bound isn't needed. Can you add [this macro](https://play.rust-lang.org/?gist=79d54732533d79bf007121bd80a6fbf3&version=nightly&mode=debug) to futures-core and use it instead?